openapiGenerator.go

```diff
@@ -390,14 +406,132 @@ func (g *openapiGenerator) generateMessageSchema(message *protomodel.MessageDesc
 	o := openapi3.NewObjectSchema()
 	o.Description = g.generateDescription(message)
 
+	oneOfFields := make(map[int32][]string)
 	for _, field := range message.Fields {
+		fieldName := g.fieldName(field)
+
+		opts, ok := proto.GetExtension(field.GetOptions(), options.E_Options).(*options.FieldOptions)
+		if ok && opts != nil {
+			fieldDesc := g.generateMultiLineDescription(field)
+			repeated := field.IsRepeated()
+
+			switch {
+			case opts.GetTypeObject():
```

Comment:
are GetTypeObject and GetTypeValue ok to be specified at the same time? 
if yes, we need to handle
if not, we need to error

i don't see these in the tests, is it worth updating the test to include them?

Reply:
Currently we give preference to the object type, much like what was done with Cue. It's not trivial to generate errors from this part of the code without a much broader refactor that I would like to avoid with this change. Option protos are generally defined flat, so defining oneofs doesn't seem to work based on my testing; may be possible but don't see documentation or examples to do so.

The only difference between these 2 is that Value doesn't add `type:object` to the schema.

Reply:
this has changed to using marker comments instead of proto options